packages/router/src/adapters/web3signer/index.ts

```diff
@@ -54,16 +54,27 @@ export class Web3Signer extends Signer {
 
   public async signTransaction(transaction: providers.TransactionRequest): Promise<string> {
     const tx = await utils.resolveProperties(transaction);
-    const baseTx: utils.UnsignedTransaction = {
-      to: tx.to || undefined,
-      nonce: tx.nonce ? BigNumber.from(tx.nonce).toNumber() : undefined,
-      gasLimit: tx.gasLimit || undefined,
-      gasPrice: tx.gasPrice || undefined,
-      data: tx.data || undefined,
-      value: tx.value || undefined,
-      chainId: tx.chainId || undefined,
-      type: tx.type || undefined,
-    };
+    const baseTx: utils.UnsignedTransaction = Object.assign(
+      {
+        to: tx.to || undefined,
+        nonce: tx.nonce ? BigNumber.from(tx.nonce).toNumber() : undefined,
+        gasLimit: tx.gasLimit || undefined,
+        data: tx.data || undefined,
+        value: tx.value || undefined,
+        chainId: tx.chainId || undefined,
+      },
+      // If an EIP-1559 transaction, use the EIP-1559 specific fields.
+      tx.type && tx.type === 2
```

Comment:
this tuple condition will be problematic if there *is* a 0 type assigned (the first `tx.type` is always false). maybe do `typeof tx.type`?

Reply:
great catch, i'll just check if it's === null or === undefined

Reply:
or just check `tx.type === 2` without the first part lol